zipkin-lens/src/prop-types/index.js

```diff
@@ -0,0 +1,77 @@
+import PropTypes from 'prop-types';
+
+export const spanTagPropTypes = PropTypes.shape({
+  key: PropTypes.string.isRequired,
+  value: PropTypes.string.isRequired,
+});
+
+export const spanTagsPropTypes = PropTypes.arrayOf(
+  spanTagPropTypes,
+);
+
+export const spanAnnotationPropTypes = PropTypes.shape({
+  value: PropTypes.string.isRequired,
+  timestamp: PropTypes.number.isRequired,
+  endpoint: PropTypes.string.isRequired,
+  relativeTime: PropTypes.string.isRequired,
+});
+
+export const spanAnnotationsPropTypes = PropTypes.arrayOf(
+  spanAnnotationPropTypes,
+);
+
+export const detailedSpanPropTypes = PropTypes.shape({
```

Comment:
maybe add a TODO as while I think your assessment is correct based on span row conversion, the underlying data only requires trace and span ID. We need to revisit things like this once UX is working as desired https://github.com/openzipkin/zipkin-api/blob/master/zipkin2-api.yaml

Reply:
@adriancole 
Sorry, I don't understand what you mean.
What should I do?

Reply:
Add above this line

```
// TODO: Verify which fields we should enforce here, as some are optional per https://github.com/openzipkin/zipkin-api/blob/master/zipkin2-api.yaml
```

Reply:
OK
Thank you!